### PR TITLE
DSND-473 - Spring Boot app initialisation

### PR DESCRIPTION
**Pull request by iichr**
- spring and dependencies initialised - application builds without errors
- unit and integrations tests can be created
  - includes the BDD testing framework and a healthcheck endpoint Cucumber test
- docker-compose with mongo as a service dependency created

Resolves:
DSND-473

### CODE REVIEW

File: src/itest/java/uk/gov/companieshouse/company/profile/CucumberFeaturesRunnerITest.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+import org.junit.runner.RunWith;
+
+@RunWith(Cucumber.class)
+@CucumberOptions(
+        features = "src/itest/resources/features",
+        plugin = {"pretty",
+                "json:target/cucumber-report.json"
+        },
+        glue = {"uk.gov.companieshouse.company.profile.steps",
+                "uk.gov.companieshouse.company.profile.configuration"})
```
Comment (iichr) [inline src/itest/java/uk/gov/companieshouse/company/profile/CucumberFeaturesRunnerITest.java:14]: `glue` needs to contain both the steps and the cucumber spring configuration package paths

File: service-map.json [hunk 2]
```diff
@@ -0,0 +1,18 @@
+{
+  "type": "Service",
+  "name": "company-profile-api",
+  "description": "Responsible for persisting and getting company profiles and informing chs-kafka-api that a resource has changed",
+  "language": "Java",
+  "integration_outgoing": [],
```
Comment (dominicst-ch) [inline service-map.json:6]:
```
"integration_outgoing": [
    "company-links-consumer,GETS_DATA_FROM",
    "chs-kafka-api,CALLS_API"
],
```

**Review by dominicst-ch — APPROVED** [review]

**Review by dominicst-ch — DISMISSED** [review]
Update service map

File: service-map.json (hunk 2 above)
Comment (iichr) [reply service-map.json]: Done

**Review by vchitti-kainos — APPROVED** [review]

**Review by zahidm3 — APPROVED** [review]

**Pull request merged**